client/jetpack-connect/persistence-utils.js

```diff
@@ -24,3 +31,34 @@ export const retrievePlan = () => {
 	const cookies = cookie.parse( document.cookie );
 	return cookies.jetpack_connect_selected_plan;
 };
+
+export const persistSession = url => {
+	const options = {
+		maxAge: JETPACK_CONNECT_TTL_SECONDS,
+		path: '/',
+	};
+	document.cookie = cookie.serialize( 'jetpack_connect_session_url', urlToSlug( url ), options );
+};
+
+export const isCalypsoStartedConnection = siteSlug => {
+	const cookies = cookie.parse( document.cookie );
+	return cookies.jetpack_connect_session_url === urlToSlug( siteSlug );
+};
+
+export const persistFlowType = flowType => {
+	if ( ! includes( FLOW_TYPES, flowType ) ) {
+		return;
+	}
+	const options = { path: '/' };
+	document.cookie = cookie.serialize( 'jetpack_connect_session_flowtype', flowType, options );
+};
+
+export const clearFlowType = () => {
+	const options = { path: '/' };
+	document.cookie = cookie.serialize( 'jetpack_connect_session_flowtype', '', options );
+};
+
+export const retrieveFlowType = () => {
+	const cookies = cookie.parse( document.cookie );
+	return cookies.jetpack_connect_session_flowtype;
+};
```

Comment:
Just an observation, repeating a lot of code here. Might be nice to abstract helpers `{get,set,clear}Cookie( name, value )` or something along those lines.